types/experiments.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * The experiment definition accessible to Firefox via Remote Settings.
+ * It is compatible with ExperimentManager.
+ */
+export interface ExperimentRecipe {
+  /** A unique identifier for the Recipe */
+  id: string;
+  /** JEXL expression defined in an Audience */
+  filter_expression: string;
```

Comment:
Nit: Should this be `filterExpression` now?

Reply:
in this case `filter_expression` has to be like this in order to maintain compatibility with the existing firefox implementation

Reply:
Interesting. I wonder if we can use https://eslint.org/docs/rules/camelcase to enforce everything uses camelcase vs underscores and we can ignore specific lines like this one. I'll play around with it today and see if that works and makes sense.

Reply:
```js
    camelcase: [
      "error",
      {
        properties: "always",
        allow: ["^experiments_", "^messaging_", "^normandy_"],
      },
    ],
```

Now, running <kbd>npm run lint</kbd> gives me the following errors/warnings:

```sh
npm run lint

> @mozilla/nimbus-shared@0.0.3-dev lint /Volumes/Dev/github/mozilla/rapid-experiments-shared
> make lint

Makefile:90: warning: overriding commands for target `&'
Makefile:75: warning: ignoring old commands for target `&'
eslint .

/Volumes/Dev/github/mozilla/rapid-experiments-shared/types/experiments.ts
   9:3   error    Identifier 'filter_expression' is not in camel case  camelcase
  80:26  warning  Unexpected any. Specify a different type             @typescript-eslint/no-explicit-any

/Volumes/Dev/github/mozilla/rapid-experiments-shared/types/messaging.ts
   1:19  error  Identifier 'string_id' is not in camel case          camelcase
  30:3   error  Identifier 'notification_text' is not in camel case  camelcase
  36:3   error  Identifier 'active_color' is not in camel case       camelcase
  37:3   error  Identifier 'heading_text' is not in camel case       camelcase
  42:3   error  Identifier 'icon_dark_theme' is not in camel case    camelcase

/Volumes/Dev/github/mozilla/rapid-experiments-shared/types/normandy.ts
  10:3  error  Identifier 'action_name' is not in camel case  camelcase

✖ 8 problems (7 errors, 1 warning)

make: *** [lint] Error 1
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @mozilla/nimbus-shared@0.0.3-dev lint: `make lint`
npm ERR! Exit status 2
```

Not sure if any of those are truly errors, or if we need some more rule filtering/overrides to allow for the underscored variables in the messaging.ts file.